### PR TITLE
[AzureSpringCloudV0] Using PUT method when deploying custom container app

### PR DESCRIPTION
**Pull request by RuoyuWang-MS**
**Task name**: <Name of changed or new pipeline task>
AzureSpringCloudV0
**Description**: <Describe your changes here>
Fix one bug when deploying custom container to Azure Spring Apps.
When deploy custom container in ASA, we should use PUT method instead of PATCH method.
Also need to keep other properties unchanged when deploying custom container app.
**Documentation changes required:** (Y/N) <Please mark if documentation changes are required>
N
**Added unit tests:** (Y/N) <Please mark if unit tests were added or updated according changes>
N
**Attached related issue:** (Y/N) <Please add link to related issue here>
Y
https://github.com/microsoft/azure-pipelines-tasks/issues/19132
**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

File: _generated/AzureSpringCloudV0_Node20/deploymentProvider/azure-arm-spring-apps.ts [hunk 1]
```diff
@@ -509,8 +509,18 @@ export class AzureSpringApps {
...
+        if (!createDeployment) {
+            deploymentResource = await this.getDeploymentInfo(appName, deploymentName);
+            isCustomContainer = deploymentResource["properties"]["source"]["type"] == SourceType.CUSTOM_CONTAINER || deploymentUpdateRequestBody["properties"]["source"]["type"] == SourceType.CUSTOM_CONTAINER;
+        }
+        let method = (createDeployment || isCustomContainer) ? 'PUT' : 'PATCH';
+        if (isCustomContainer) {
+            const deploymentSettings = {...deploymentResource["properties"]["deploymentSettings"], ...deploymentUpdateRequestBody["properties"]["deploymentSettings"]};
+            deploymentUpdateRequestBody["properties"]["deploymentSettings"] = deploymentSettings;
+            deploymentUpdateRequestBody["sku"] = deploymentResource["sku"];
+        }
```
Comment (allxiao) [inline _generated/AzureSpringCloudV0_Node20/deploymentProvider/azure-arm-spring-apps.ts:523]: It looks like the changes to all the 3 files are merely the same. Can we extract the logic to a common place and just reference it in these places?

File: Tasks/AzureSpringCloudV0/deploymentProvider/azure-arm-spring-apps.ts [hunk 2]
```diff
@@ -509,8 +509,18 @@ export class AzureSpringApps {
...
         console.log(`${createDeployment ? 'Creating' : 'Updating'} ${appName}, deployment ${deploymentName}...`);
-
-        let method = createDeployment ? 'PUT' : 'PATCH';
+        let isCustomContainer = false;
+        let deploymentResource = {};
+        if (!createDeployment) {
+            deploymentResource = await this.getDeploymentInfo(appName, deploymentName);
+            isCustomContainer = deploymentResource["properties"]["source"]["type"] == SourceType.CUSTOM_CONTAINER || deploymentUpdateRequestBody["properties"]["source"]["type"] == SourceType.CUSTOM_CONTAINER;
+        }
+        let method = (createDeployment || isCustomContainer) ? 'PUT' : 'PATCH';
```
Comment (allxiao) [inline Tasks/AzureSpringCloudV0/deploymentProvider/azure-arm-spring-apps.ts:518]: per talk offline, please also check if we need to update for other source types.

Comment (RuoyuWang-MS) [reply Tasks/AzureSpringCloudV0/deploymentProvider/azure-arm-spring-apps.ts:518]: I have tried Jar/Source in Standard tier and I can use PATCH to deploy Source to a Jar app.
For custom container cases, we have a validator logic in RP deployment controller to reject the PATCH request about byoc cases and throw `properties.source.type can only be changed in PUT request` message

File: _generated/AzureSpringCloudV0_Node20/deploymentProvider/azure-arm-spring-apps.ts (hunk 1 above)
Comment (RuoyuWang-MS) [reply _generated/AzureSpringCloudV0_Node20/deploymentProvider/azure-arm-spring-apps.ts:523]: The two files in `_generated` folder are build result of `node make.js build --task AzureSpringCloudV0` command

Comment (allxiao) [reply _generated/AzureSpringCloudV0_Node20/deploymentProvider/azure-arm-spring-apps.ts:523]: as discussed, this is added during the Node 20 migration.

**Review by allxiao — APPROVED** [review]

**Comment by RuoyuWang-MS** [comment]
@mmrazik Please help merge this PR at your convenience. Thanks a lot!

**Comment by RuoyuWang-MS** [comment]
@mmrazik Would you please help merge this PR at your convenience? This PR fix a customer reported issue: https://github.com/microsoft/azure-pipelines-tasks/issues/19132. Thanks a lot!

**Comment by anishp55** [comment]
Could we please get this merged?  It would be nice to have this working during demos.

**Review by DmitriiBobreshev — APPROVED** [review]

**Review by max-zaytsev — APPROVED** [review]

**Comment by merlynomsft** [comment]
@microsoft/azure-spring-apps @RuoyuWang-MS @anishp55 - we gave you write access to the repo so you can merge PRs for your tasks directly.

**Pull request merged**